### PR TITLE
Bugfix/white238/fixgitlabdepbuild

### PR DESCRIPTION
**Pull request by white238**
# Summary

- This PR is a bugfix
- It does the following (modify list as needed):
  - Updates BLT so that we can convert CTest output to JUnit
  - Export bare minimum of imported targets required to use new blt_import_library used in new BLT
  - Add mirror to build_tpls (fixes deps build that runs after PR merge)
  - Convert more python print commands

### CODE REVIEW

**Comment by white238** [comment]
LGTM

File: scripts/llnl_scripts/llnl_lc_build_tools.py [hunk 1]
```diff
@@ -259,16 +261,16 @@ def uberenv_create_mirror(prefix, project_file, mirror_path):
...
-    print "[~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~]"
-    print "[ End of expected warnings from 'spack --create-mirror'                        ]"
-    print "[~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~]"
+    print("[~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~]")
+    print("[ End of expected warnings from 'spack --create-mirror'                        ]")
+    print("[~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~]")
     set_axom_group_and_perms(mirror_path)
     return res
-
+`
```
Comment (kennyweiss) [inline scripts/llnl_scripts/llnl_lc_build_tools.py:273]: Remove me please!

**Comment by white238** [comment]
Bumping BLT caused the fixing of fortran openmp flags to not trigger.  This is currently broken. so sad.

**Comment by joshessman-llnl** [comment]
LGTM

**Review by agcapps — APPROVED** [review]
Thanks Chris!

**Comment by joshessman-llnl** [comment]
LGTM

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -78,7 +78,7 @@ stages:
     - echo -e "section_start:$(date +%s):build_and_test\r\e[0K
       Build and test ${CI_PROJECT_NAME}"
     # Autodetects host-config created by build_tpls
-    - lalloc 1 -W 20 ${LSF_ACCOUNT} scripts/llnl_scripts/build_src.py -v --automation
+    - lalloc 1 -W 25 ${LSF_ACCOUNT} scripts/llnl_scripts/build_src.py -v --automation
```
Comment (white238) [inline .gitlab-ci.yml:81]: We were apparently too close to the limit. It was occasionally failing due to time limit

File: scripts/llnl_scripts/build_src.py [hunk 3]
```diff
@@ -89,7 +89,7 @@ def main():
         # Default to build all SYS_TYPE's host-configs in host-config/
         build_all = not opts["hostconfig"] and not opts["automation"]
         if build_all:
-            res = build_and_test_host_configs(repo_dir, timestamp, False, opts["verbose"])
+            res = build_and_test_host_configs(repo_dir, job_name, timestamp, False, opts["verbose"])
```
Comment (white238) [inline scripts/llnl_scripts/build_src.py:92]: Pretty sure this was the cause of the weird email report this morning.

**Comment by white238** [comment]
LGTM

File: .gitlab-ci.yml (hunk 2 above)
Comment (kennyweiss) [reply .gitlab-ci.yml:81]: Any downside to raising it higher -- e.g. 40 mins? to avoid unnecessary timeouts?

**Review by kennyweiss — APPROVED** [review]
Thanks!

**Comment by white238** [comment]
Not sure why i cant reply to your above comment.  I have noticed that the higher i put the time limit the longer it takes to schedule.  I think there is a <30 minute limit to some of the queue.  I could be wrong and this could be completely anecdotal.

**Pull request merged**